plugin/Utility/RedirectIfOldUrl.php

```diff
@@ -5,34 +5,38 @@
 use DI\ContainerBuilder;
 use onOffice\WPlugin\Controller\EstateDetailUrl;
 use onOffice\WPlugin\WP\WPPageWrapper;
+use onOffice\WPlugin\WP\WPRedirectWrapper;
 
 class RedirectIfOldUrl
 {
 	/** @var EstateDetailUrl */
 	private $_pLanguageSwitcher;
+	private $_wpPageWrapper;
+	private $_wpRedirectWrapper;
 
 	public function __construct()
 	{
 		$pContainerBuilder = new ContainerBuilder;
 		$pContainerBuilder->addDefinitions(ONOFFICE_DI_CONFIG_PATH);
 		$pContainer = $pContainerBuilder->build();
 		$this->_pLanguageSwitcher = $pContainer->get(EstateDetailUrl::class);
+		$this->_wpPageWrapper = $pContainer->get(WPPageWrapper::class);
```

Comment:
It'd be cleaner and easier to test if the dependencies were obtained as constructor parameters:

```php
	public function __construct(EstateDetailUrl $pEstateDetailUrl, WPPageWrapper $pWPPageWrapper, WPRedirectWrapper $pWPRedirectWrapper)
	{
		$this->_pLanguageSwitcher = $pEstateDetailUrl;
		$this->_wpPageWrapper = $pWPPageWrapper;
		$this->_wpRedirectWrapper = $pWPRedirectWrapper;
	}
```

Reply:
hi @jayay,
I need setter functions to set mock for the test

Reply:
Just use the constructor! That'd be a lot cleaner.

Reply:
@jayay constructor accepts params on the correct instance. eg WPPageWrapper. But when I want to pass the mock of WPPageWrapper, eg WPPageWrapperMocked, the constructor gives an error

Reply:
@jayay constructor accepts params on the correct instance. eg WPPageWrapper. But when I want to pass the mock of WPPageWrapper, eg WPPageWrapperMocked, the constructor gives an error

Reply:
It should work, as long as the new mock is a subclass of `WPPageWrapper`. Mocks in PHPUnit inherit from the mocked class. What exactly is the error?